### DISCUSSION TITLE
rtl support without config in tailwind.config.js

### DISCUSSION THREAD

**Discussion by ziaadini** [saadeghi/daisyui#2141] [Ideas / feature requests]
hi, tailwind suports [rtl](https://tailwindcss.com/blog/tailwindcss-v3#rtl-and-ltr-modifiers) in version 3, and in the otherhand there is a problem in multilanguage apps they can not change config in run time, at least i didn't find anything.
it would be nice to use tailwind rtl support without any extra configuration.
thanks pouya for your perfect plugin.

**Comment by saadeghi** [comment]
Currently we're using `rtlcss` package to make the RTL style. I'm planning to use logical properties so I won't need that package for rotating margins, paddings, etc. But some styles like box shadows will be a challenge that way.  
I'm working on it and hopefully I will find a way to make it work without config change for the next major version of daisyUI.

**Comment by ziaadini** [comment]
what about using `tailwindcss-flip` is it necessary for the next major version?
we are wrapping all components using `nuxt-layers` and it's important for our projects.
thanks

**Reply by saadeghi** [reply]
`tailwindcss-flip` is really helpful for automatically rotating any Tailwind utility class. It turns `mr-6` to margin-left so there would be no need to manually define RTL alternative class for to all the Tailwind utility classes.

So if anyone wants that to be automated they can use `tailwindcss-flip` otherwise they should use `rtl:` `ltr:` prefixes manually.

**Reply by ziaadini** [reply]
okay thanks, for the problem in this version what is the solution to toggle between rtl and ltr at runtime?

**Reply by saadeghi** [reply]
You can generate 2 CSS files. It's more efficient to have 2 different CSS files for LTR and RTL because the end user doesn't switch between LTR/RTL regularly. They only need to choose it once and one CSS file that only contains style for LTR or RTL is going to be smaller than a CSS file that contains bidirectional styles for both (and one of the directions will be useless for the end user)

Here's an example setup of having 2 tailwind.config.js files to generate 2 CSS files:  
https://github.com/saadeghi/daisyui-bidirectional-builds

Let me know if you have any questions.
